espnet/espnet.c

```diff
@@ -299,39 +330,48 @@ static unsigned long _random(void) {
 	return 0;
 }
 // #if CONFIG_IDF_TARGET_ESP32S2 || CONFIG_IDF_TARGET_ESP32S3 || CONFIG_IDF_TARGET_ESP32C3
-//     uint32_t (* _slowclk_cal_get(void)
+//	 uint32_t (* _slowclk_cal_get(void)
 // #endif
 static void _log_write(uint32_t level, const char* tag, const char* format, ...) {
-	P(_log_write)
+	va_list argList;
+	printf("[%s] ", tag);
+	va_start(argList, format);
+	printf(format, argList);
+	va_end(argList);
+	printf("\n");
 }
 static void _log_writev(uint32_t level, const char* tag, const char* format, va_list args) {
-	P(_log_writev)
+	printf("[%s] ", tag);
+	printf(format, args);
+	printf("\n");
 }
+
 static uint32_t  _log_timestamp(void) {
 	P(_log_timestamp)
 	return 0;
 }
 static void* _malloc_internal(size_t size) {
-	P(_malloc_internal)
-	return NULL;
+	printf("called: _malloc_internal(%d)\n", size);
+	return malloc(size);
```

Comment:
Note that simply calling `malloc` may not be enough. I believe only some memory supports DMA. But it should be good enough for now.